### DISCUSSION TITLE
wlroots: Use native libraries?

### DISCUSSION THREAD

**Discussion by bugi** [Homebrew/discussions#4095] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I've started work on packaging [libwlroots](https://gitlab.freedesktop.org/wlroots/wlroots).  I have some questions related to library sourcing.

One place I looked was the [mesa formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/mesa.rb).  It provides its own copies of EGL, GLESv2, and GBM.  It that the norm?  Should I just set mesa as a dependency of libwlroots?

I haven't found brew-packaged libinput, udev, and some other dependencies.  Is it acceptable to just assume that builders and users have those installed?  If so, should I communicate that to the user somehow, like "hey you, run sudo apt install foo"?

This is my first attempt to package something for Homebrew.

**Comment by SMillerDev** [comment, marked as answer]
No, anything that you depend on for the formula should be declared with a `depends_on` statement.
